service/internal/checksum/middleware_compute_input_checksum.go

```diff
@@ -130,29 +129,15 @@ func (m *computeInputPayloadChecksum) HandleFinalize(
 		})
 	}()
 
-	// If no algorithm was specified, and the operation requires a checksum,
-	// fallback to the legacy content MD5 checksum.
-	algorithm, ok, err = getInputAlgorithm(ctx)
-	if err != nil {
-		return out, metadata, err
-	} else if !ok {
-		if m.RequireChecksum {
-			checksum, err = setMD5Checksum(ctx, req)
-			if err != nil {
-				return out, metadata, computeInputHeaderChecksumError{
-					Msg: "failed to compute stream's MD5 checksum",
-					Err: err,
-				}
-			}
-			algorithm = Algorithm("MD5")
-		}
-		return next.HandleFinalize(ctx, in)
-	}
-
 	// If the checksum header is already set nothing to do.
-	checksumHeader := AlgorithmHTTPHeader(algorithm)
-	if checksum = req.Header.Get(checksumHeader); checksum != "" {
-		return next.HandleFinalize(ctx, in)
+	for _, supportedAlgorithm := range supportedAlgorithms {
+		header := AlgorithmHTTPHeader(supportedAlgorithm)
+		if cs := req.Header.Get(header); cs != "" {
+			// need to add header value to metadata
+			algorithm = supportedAlgorithm
```

Comment:
~~Are these two values doing anything? You return right after setting them.~~ Nevermind, I see the defer() above doing stuff with them.